src/ert/_c_wrappers/enkf/config/gen_kw_config.py

```diff
@@ -71,53 +32,57 @@ def __init__(
         if not os.path.isfile(parameter_file):
             raise IOError(f"No such file:{parameter_file}")
 
-        c_ptr = self._alloc_empty(key, tag_fmt)
-        if c_ptr:
-            super().__init__(c_ptr)
-        else:
-            raise ValueError(
-                "Could not instantiate GenKwConfig with "
-                f'key="{key}" and tag_fmt="{tag_fmt}"'
-            )
-        self._set_parameter_file(parameter_file)
-        self._set_template_file(template_file)
+        self._key = key
```

Comment:
very much details - but - as i see it the only consumer of `self._key` is `getKey(self)...` , and in the code now it is also said `self.name = key` - which in my head says `self.name == self._key` . Are there any reasons for having the `self._key`?  I.e - could `getKey(..)` function return `self.name` ?

Reply:
Yes. Let's do that. :+1: